pkg/collector/corechecks/system/disk/disk/disk.go

```diff
@@ -148,39 +262,412 @@ func (c *Check) instanceConfigure(data integration.Data) error {
 			}
 		}
 	}
+	tagByLabel, found := unmarshalledInstanceConfig["tag_by_label"]
+	if tagByLabel, ok := tagByLabel.(bool); found && ok {
+		c.cfg.tagByLabel = tagByLabel
+	}
+	useLsblk, found := unmarshalledInstanceConfig["use_lsblk"]
+	if useLsblk, ok := useLsblk.(bool); found && ok {
+		c.cfg.useLsblk = useLsblk
+	}
+	blkidCacheFile, found := unmarshalledInstanceConfig["blkid_cache_file"]
+	if blkidCacheFile, ok := blkidCacheFile.(string); found && ok {
+		c.cfg.blkidCacheFile = blkidCacheFile
+	}
+	if c.cfg.tagByLabel && c.cfg.useLsblk && c.cfg.blkidCacheFile != "" {
+		return errors.New("Only one of 'use_lsblk' and 'blkid_cache_file' can be set at the same time.")
+	}
+	serviceCheckRw, found := unmarshalledInstanceConfig["service_check_rw"]
+	if serviceCheckRw, ok := serviceCheckRw.(bool); found && ok {
+		c.cfg.serviceCheckRw = serviceCheckRw
+	}
 
+	createMounts, found := unmarshalledInstanceConfig["create_mounts"]
+	if createMounts, ok := createMounts.([]interface{}); found && ok {
+		for _, createMount := range createMounts {
+			var m Mount
+			err = mapstructure.Decode(createMount, &m)
```

Comment:
Why are we using `mapstructure` instead of `yaml`?

Reply:
`createMount` comes from `yaml.Unmarshal` but was decoded into map[string]interface{} instead of a struct.